### DISCUSSION TITLE
Returning the URL if the shortened version already exists?

### DISCUSSION THREAD

**Discussion by dafacto** [YOURLS/YOURLS#3511] [General]
Hello,

I have a script that creates short URLs, and then copies the shortened version to the clipboard on my Mac. Often, I'll need the short version of a URL that already exists in my Yourls database. In the past, Yourls returned the existing shortened version. With some recent update that I've made to my installation, it's now returning an error instead.

Is there a configuration option that would allow me to have my installation revert to the previous behavior, i.e. returning the shortened version of a URL that already exists in my database?

**Comment by adnaanqureshi** [comment]
i also am struggling with perhaps the same issue. when using the api to shorten links and with unique links set to true (as default) the xml returned for an existing longurl is structurally different to the xml returned for a new link. this means my code now has to process this. i’m sure before it would just return the existing shorturl in a similar xml structure as it would for a new url

**Comment by dafacto** [comment]
I just discovered the option to NOT have unique short URLs, which fixes my problem (i.e. my script isn't breaking any longer). For my purposes, I don't need unique URLs. It would be nice, though, to have unique URLs and that YOURLS simply returns the existing URL if one exists.

**Reply by adnaanqureshi** [reply]
having unique urls true doesn’t break anything - but the xml you get back from the api contains an error and it also contains the existing short url. you just need to process this in your app. but i agree it should just return the existing shorturl without errors in the xml
UPDATE - There is an error in this post - You do NOT get the old short url returned from the API for an existing long url, when unique urls is true

**Reply by dafacto** [reply]
I'm using the JSON return form, and my script is getting web server errors back when I try to shorten an existing URL. Also, if I use the web UI to shorten an existing URL, it returns an error in the UI.

**Reply by adnaanqureshi** [reply]
right. i have questions then. how are you getting a json? i’m using the https api which returns an xml. also, in your api response , is there any details of the error?

**Reply by dafacto** [reply]
format=json. I've already changed my configuration, so I'm not seeing errors to inspect for details. :(

**Reply by adnaanqureshi** [reply]
oh i didn’t realise you could change the format. i’m going to experiment later - i’m quite sure yesterday when i had unique urls true, i was getting a response with an error node and then another node with the old shorturl (EDITED after testing - this is incorrect- the old short url is NOT returned)

**Reply by dafacto** [reply]
I tried to find it in the API docs to point you to, but I must be blind, as I can't find the API interface definition in the docs.

**Reply by ozh** [reply]
Still something to migrate to the newish online documentation, but provided since the beginning with the readme that ships with YOURLS (eg https://yourls.org/readme.html#API)

**Reply by adnaanqureshi** [reply]
thanks ozh. 

so ive tested now and can confirm that with unique urls set to TRUE, if you send an existing long url to the api (with default xml format) you get back bool(false). 

so - my question is, is there a plugin that will just return the existing short url  when the long url already exists ?

as defacto, ive simply set unique urls to false to overcome this as a workaround.

**Reply by dgw** [reply]
> is there a plugin that will just return the existing short url when the long url already exists ?

There is https://github.com/adigitalife/yourls-change-error-messages which is likely untested with current YOURLS releases (it hasn't been updated in 5 years).

**Reply by adnaanqureshi** [reply]
this looks exactly like what i was after. thank you.
